build/source/driver/multi_driver.f90

```diff
@@ -850,12 +851,14 @@ program multi_driver
    allocate(zSoilReverseSign(gru_struc(iGRU)%hruInfo(iHRU)%nSoil),stat=err); call handle_err(err,'problem allocating space for zSoilReverseSign')
    zSoilReverseSign(:) = -progStruct%gru(iGRU)%hru(iHRU)%var(iLookPROG%iLayerHeight)%dat(gru_struc(iGRU)%hruInfo(iHRU)%nSnow+1:nLayers)
   
+   nSoil = 9999
+
    ! get NOAH-MP parameters
    call REDPRM(typeStruct%gru(iGRU)%hru(iHRU)%var(iLookTYPE%vegTypeIndex),      & ! vegetation type index
                typeStruct%gru(iGRU)%hru(iHRU)%var(iLookTYPE%soilTypeIndex),     & ! soil type
                typeStruct%gru(iGRU)%hru(iHRU)%var(iLookTYPE%slopeTypeIndex),    & ! slope type index
                zSoilReverseSign,                                                & ! * not used: height at bottom of each layer [NOTE: negative] (m)
-               gru_struc(iGRU)%hruInfo(iHRU)%nSoil,                             & ! number of soil layers
+               nSoil,                                                           & ! number of soil layers
```

Comment:
Need a comment to say passing a `maxSoilLayer` in order to artificially pass the check with `nroot`, that this is done in order to avoid making any changes to the `noah-mp` code, and the workaround is appropriate because `nroot` from the Noah-Mp veg tables (as read here) is not used in summa.

Reply:
Comment has been added.